buildSrc/src/main/java/org/opensearch/gradle/testclusters/TestClusterConfiguration.java

```diff
@@ -107,6 +107,8 @@ public interface TestClusterConfiguration {
 
     void setPreserveDataDir(boolean preserveDataDir);
 
+    void setHttpProtocol(String protocol);
```

Comment:
Previously dB had suggested swapping it from just being based on HTTP to allowing a more generic string entry: https://github.com/opensearch-project/OpenSearch/pull/8900#pullrequestreview-1550809454

Reply:
Sure, but there are 2 part to that:
 - protocol (let say, http/1.1, http/2, http/3, grpc)
 - TLS / DTLS

```
http/1.1 + TLS = https
http/2 + TLS = https
http/3 + TLS = https
```

So we need two properties here I think: protocol + security.

Reply:
Sorry, I am not sure I follow. Does providing the end result (https or http) not allow for the same configuration? Sorry if I am misunderstanding.

Reply:
Does the swap I made address your concern? Letting us input a string allows for more configuration in how we want to test things it seems. For instance we can confirm DTLS and TLS work the same but may bea ble to later add tests that examine how they differ. This seems like what you were suggesting originally.

Reply:
> Does the swap I made address your concern?

It goes towards right direction for sure but since this is a string, what value(s) should be acceptable there? (boolean flag is significantly easier to reason about)

Reply:
I will swap it to a boolean but you won't be able to swap between UDP and TCP.